ihatemoney/models.py

```diff
@@ -263,6 +271,37 @@ def get_pretty_bills(self, export_format="json"):
             )
         return pretty_bills
 
+    def switch_currency(self, new_currency):
+        # Update converted currency
+        if new_currency != self.default_currency:
+            if (
+                new_currency == CurrencyConverter.no_currency
+                and self.has_multiple_currencies()
+            ):
+                raise ValueError(f"Can't unset currency of project {self.id}")
+            for bill in self.get_bills_unordered():
+
+                if new_currency == CurrencyConverter.no_currency:
+                    # Use old currency to flatten all amount before stripping
+                    bill.converted_amount = CurrencyConverter().exchange_currency(
+                        bill.amount, bill.original_currency, self.default_currency
+                    )
+                    # Strip currency
+                    bill.amount = bill.converted_amount
+                    bill.original_currency = CurrencyConverter.no_currency
+                else:
+                    # Switch to new currency for everyone
+                    bill.converted_amount = CurrencyConverter().exchange_currency(
+                        bill.amount, bill.original_currency, new_currency
+                    )
+                    # Add the currency for bills that were created without currency
+                    if bill.original_currency == CurrencyConverter.no_currency:
+                        bill.original_currency = new_currency
```

Comment:
Is it even possible to have a bill with no currency here?

It looks like the two changes are in the wrong order: to compute `bill.converted_amount`, we need `bill.original_currency`, but it is only set in the second change.

Reply:
Yes, it is possible, when switching from XXX (no currency) -> EUR. `original_currency` is XXX, and we must set it to the new default currency.
If switching USD -> EUR, this is not possible.

The two affectations does the following:
- Compute the *new* `converted_amount` by using the conversion from bill's `original_currency` to `new_currency`.
- Override the `original_currency` if and only if it was set to XXX.

So the first affectation will be a "simple" recomputation, since conversion ratio of XXX is set to 1.

Reply:
Ah, right, it makes sense with the additional knowledge that the conversion rate of XXX is 1 :)

It would clarify the code to make that explicit:

- if original_currency == XXX, just set `converted_amount = amount` and `original_currency = new_currency`
- otherwise, apply the conversion

Reply:
OK done